app/controllers/admin/volunteers.py

```diff
@@ -111,16 +112,16 @@ def volunteerDetailsPage(eventID):
     if not (g.current_user.isCeltsAdmin or (g.current_user.isCeltsStudentStaff and g.current_user.isProgramManagerForEvent(event))):
         abort(403)
 
-    eventRsvpData = list(EventRsvp.select().where(EventRsvp.event==event))
-    eventParticipantData = list(EventParticipant.select().where(EventParticipant.event==event))
+    eventRsvpData = list(EventRsvp.select(EmergencyContact, EventRsvp).join(EmergencyContact, JOIN.LEFT_OUTER, on=(EmergencyContact.user==EventRsvp.user)).where(EventRsvp.event==event))
+    eventParticipantData = list(EventParticipant.select(EmergencyContact, EventParticipant).join(EmergencyContact, JOIN.LEFT_OUTER, on=(EmergencyContact.user==EventParticipant.user)).where(EventParticipant.event==event))
     participantsAndRsvp = (eventParticipantData + eventRsvpData)
-    eventParticipantUsers = [obj.user for obj in eventParticipantData]
-    eventNonAttendedData = [obj.user for obj in eventRsvpData if obj.user not in eventParticipantUsers]
+    eventParticipantUsers = [obj for obj in eventParticipantData]
+    eventNonAttendedData = [obj for obj in eventRsvpData if obj not in eventParticipantUsers]
     
     #get unique list of users for each category waitlist/notwaitlist,rsvped/attended
-    waitlistUser = list(set([obj.user for obj in participantsAndRsvp if obj.rsvpWaitlist]))
-    rsvpUser = list(set([obj.user for obj in eventRsvpData if not obj.rsvpWaitlist ]))
-    attendedUser= list(set([obj.user for obj in eventParticipantData if obj.user not in eventNonAttendedData]))
+    waitlistUser = list(set([obj for obj in participantsAndRsvp if obj.rsvpWaitlist]))
+    rsvpUser = list(set([obj for obj in eventRsvpData if not obj.rsvpWaitlist ]))
+    attendedUser= list(set([obj for obj in eventParticipantData if obj not in eventNonAttendedData]))
```

Comment:
We're mixing `EventRsvp` and `EventParticipant` now instead of using all User objects. That's fine if everything still works, but this variable arrangement doesn't completely make sense anymore if they aren't all lists of users. For example, line 118 is an unnecessary list comprehension, because the two lists are the same now. Also, for line 124, `attendedUser` will always be the same as `eventParticipantData`, because there will never be `EventParticipant` objects in `eventNonAttendedData`.